plugins/workflow/category/src/Extension/Category.php

```diff
@@ -0,0 +1,336 @@
+        $section   = $event->getArgument('section');
+
+        // We need the single model context for checking for workflow
+        $singularsection = Inflector::singularize($section);
+
+        if (!$this->isSupported($component . '.' . $singularsection)) {
+            return;
+        }
+
+        $js = "
+            document.addEventListener('DOMContentLoaded', function()
+            {
+                var dropdown = document.getElementById('toolbar-status-group');
+                if (!dropdown){
+                    return;
+                }
+                var batchButton = document.getElementById('status-group-children-batch');
+                if (batchButton){
+                    batchButton.addEventListener('click', function()
+                    {
+                        var observer = new MutationObserver(function(mutations, observer) {
+                            var categorySelector = document.getElementById('batch-category-id');
+                            if (categorySelector) {
+                                categorySelector.disabled = true;
+                                observer.disconnect();
+                            }
+                        });
+
+                        observer.observe(document.body, { childList: true, subtree: true });
+                    });
+                }
+            });
+        ";
+
+        $this->getApplication()->getDocument()->addScriptDeclaration($js);
+    }
+
+    /**
+     * Method to handle the workflow transition event.
+     *
+     * @param   WorkflowTransitionEvent  $event  The event object
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function onWorkflowAfterTransition(WorkflowTransitionEvent $event): void
+    {
+        $context       = $event->getArgument('extension');
+        $extensionName = $event->getArgument('extensionName');
+        $transition    = $event->getArgument('transition');
+        $pks           = $event->getArgument('pks');
+
+        if (!$this->isSupported($context)) {
+            return;
+        }
+
+        $component = $this->getApplication()->bootComponent($extensionName);
+
+        $categoryId = $transition->options->get('category_id', 0);
+
+        if (!is_numeric($categoryId)) {
+            return;
+        }
+
+        $app = $this->getApplication();
+
+        if (!\is_object($transition) || !($transition->options instanceof Registry)) {
+            $app->enqueueMessage('PLG_WORKFLOW_CATEGORY_INVALID_TRANSITION');
+            return;
+        }
+
+
+
+        if (empty($pks) || !\is_array($pks)) {
+            $app->enqueueMessage(Text::_('PLG_WORKFLOW_CATEGORY_NO_PRIMARY_KEY'), 'error');
+            return;
+        }
+
+        $errors = 0;
+
+        foreach ($pks as $pk) {
+            if (!self::processArticle($pk, $categoryId)) {
+                $errors++;
+            }
+        }
+
+        if ($errors > 0) {
```

Comment:
I see what you did here, it's very hard to fix it, if you don't get a proper error message